DylibBinder/StreamWriterExtensions.cs

```diff
@@ -0,0 +1,887 @@
+		{
+			if (signature == null)
+				return null;
+
+			var returnType = signature.ReturnType;
+			if (returnType != null && !String.IsNullOrEmpty (returnType.ToString ()) && returnType.ToString () != "()") {
+				var enhancedReturn = StringBuilderHelper.EnhanceReturn (signature, depth, genericArgumentsCount, genericParameterCount);
+				var escapedReturn = StringBuilderHelper.EscapeCharacters (enhancedReturn);
+				if (escapedReturn != null) {
+					sw.WriteTypeValue ("returnType", escapedReturn);
+					var retList = new List<string> ();
+					if (escapedReturn.Contains ("T0"))
+						retList.Add ("T0");
+					if (escapedReturn.Contains ("T1"))
+						retList.Add ("T1");
+					return retList;
+				}
+			} else {
+				sw.WriteTypeValue ("returnType", "()");
+			}
+			return null;
+		}
+
+		static void WriteConstantAttributes (this StreamWriter sw)
+		{
+			//elements not present
+			sw.WriteTypeValue ("accessibility", "Public");
+			sw.WriteTypeValue ("isProperty", "False");
+			sw.WriteTypeValue ("isFinal", "False");
+			sw.WriteTypeValue ("isDeprecated", "False");
+			sw.WriteTypeValue ("isUnavailable", "False");
+			sw.WriteTypeValue ("isOptional", "False");
+			sw.WriteTypeValue ("isRequired", "False");
+			sw.WriteTypeValue ("isConvenienceInit", "False");
+			sw.WriteLine (" objcSelector=\"\">");
+		}
+
+		static void WritePropertyMethodGenerics (this StreamWriter sw, List<string> propertyGenericChecker, int depth, List<string> methodGenericChecker, List<string> childrenGenerics = null)
+		{
+			var genericCounter = 0;
+
+			for (int i = 0; i < 6; i++) {
+				if (propertyGenericChecker != null) {
+					if (propertyGenericChecker.Contains ($"T{i}")) {
+						genericCounter++;
+						continue;
+					}
+				}
+
+				if (methodGenericChecker != null) {
+					if (methodGenericChecker.Contains ($"T{i}")) {
+						genericCounter++;
+						continue;
+					}
+				}
+
+				if (childrenGenerics != null) {
+					if (childrenGenerics.Contains ($"T{i}")) {
+						genericCounter++;
+						continue;
+					}
+				}
+			}
+
+			sw.WriteTypeDeclarationGenericParameters (genericCounter, depth);
+		}
+
+		static void WriteTypeDeclarationGenericParameters (this StreamWriter sw, int genericsCount, int depth)
+		{
+			if (genericsCount == 0 || depth > 0) {
+				return;
+			}
+			sw.WriteLineWithIndent ("<genericparameters>");
+			Indent ();
+			for (int i = 0; i < genericsCount; i++) {
+				sw.WriteLineWithIndent ($"<param name=\"T{i}\"/>");
+			}
+			Exdent ();
+			sw.WriteLineWithIndent ("</genericparameters>");
+		}
+
+		static void WriteInnerGenericParamters (this StreamWriter sw, SwiftReflector.SwiftBaseFunctionType signature, SwiftReflector.SwiftPropertyType property, int depth)
+		{
+			var genparams = 0;
+
+			if (signature != null) {
+				// this signature contains generic parameters at the typedeclaration and function level
+				if (signature.ToString ().Contains ("0,") && signature.ToString ().Contains ("1,")) {
```

Comment:
I don't think you want to try to parse out generics like this. In general the `ToString` methods here are for debugging: you can examine a type a see what it represents easily. It also has some minor bugs that I've never gotten around to fixing.
Instead, you should decompose the type to find generics.